tests/test_integration.py

```diff
@@ -1255,6 +1257,37 @@ async def test_info_unsuccessful_start(succeed_immediately):
     assert "failed to start" in stderr.lower(), stderr
 
 
+@pytest.mark.asyncio
+async def test_filter_implementations_no_arguments():
+    output_buffer = BytesIO()
+    stdout, stderr = [], ""
+
+    def read(fd):
```

Comment:
This looks too low-level, I'd expect something to exist which does it already, but I'll have to look what that is.

Reply:
I'll have a look

Reply:
I remember now why I had tried doing it this way. The return type of `pty.spawn` ig if I am not wrong is the `pid` and not the `fd (file_descriptor)`. So `pty.spawn` requires a callback which gets the passed `file_descriptor` and when I had tried to not use `select.select` (polling) earlier, for some odd reason the callback was only reading partial data and always there was some missing data. That's why I tried introducing polling using `select.select` to wait for a split second until it gets the data and only then write to the buffer. This way there was no case of any missed data. I had also tried it using `pty.fork` and if I am not wrong and recalling correctly in that case there was no need of polling (I will try it again).

I might be wrong here / there might be some other way but whenever I tried with `pty.spawn` and no polling it was resulting in missing data :(

Reply:
Try using the [pexpect module](https://pexpect.readthedocs.io/en/stable/overview.html) with `.expect(pexpect.EOF)` I think (add that to our test requirements).

Reply:
Okay! I will give it a try 👍